### PR TITLE
refactor/PRO-1101: rid of swapping minimum swap amount

### PR DESCRIPTION
**Pull request by RGafiyatullin**
# Pull Request

Closes: PRO-1101

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have updated documentation where appropriate.

## Summary

*TBD*

### CODE REVIEW

**Review by acdibble — APPROVED** [review]
good from product side

File: state-chain/node/src/chain_spec.rs [hunk 1]
```diff
@@ -459,7 +456,8 @@ fn testnet_genesis(
 	penalties: Vec<(Offence, (i32, BlockNumber))>,
 	keygen_ceremony_timeout_blocks: BlockNumber,
 	threshold_signature_ceremony_timeout_blocks: BlockNumber,
-	minimum_swap_amounts: Vec<(assets::any::Asset, AssetAmount)>,
+	// // FIXME: unused
+	// _minimum_swap_amounts: Vec<(assets::any::Asset, AssetAmount)>,
```
Comment (dandanlen) [inline state-chain/node/src/chain_spec.rs:460]: Can be deleted.

File: state-chain/pallets/cf-swapping/src/benchmarking.rs [hunk 2]
```diff
@@ -77,13 +69,13 @@ benchmarks! {
...
-			amount: 1u128,
-		};
-		call.dispatch_bypass_filter(gov_origin)?;
+		// // reduce minimum swap amount
+		// let gov_origin = T::EnsureGovernance::try_successful_origin().unwrap();
+		// let call = Call::<T>::set_minimum_swap_amount{
+		// 	asset: Asset::Usdc,
+		// 	amount: 1u128,
+		// };
+		// call.dispatch_bypass_filter(gov_origin)?;
```
Comment (dandanlen) [inline state-chain/pallets/cf-swapping/src/benchmarking.rs:78]: Same, can be deleted.

File: state-chain/pallets/cf-swapping/src/lib.rs [hunk 3]
```diff
@@ -380,22 +362,17 @@ pub mod pallet {
...
 	}
 
 	#[pallet::genesis_build]
 	impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
-		fn build(&self) {
-			for (asset, min) in &self.minimum_swap_amounts {
-				MinimumSwapAmount::<T>::insert(asset, min);
-			}
-		}
+		fn build(&self) {}
```
Comment (dandanlen) [inline state-chain/pallets/cf-swapping/src/lib.rs:370]: I'm pretty sure we can just delete the GenesisConfig struct and associated impls entirely.

File: state-chain/pallets/cf-swapping/src/lib.rs [hunk 4]
```diff
@@ -740,14 +695,15 @@ pub mod pallet {
...
-				return Err(CcmFailReason::PrincipalSwapAmountTooLow)
 			}
+			// else if source_asset != destination_asset &&
+			// 	!principal_swap_amount.is_zero() &&
+			// 	principal_swap_amount < MinimumSwapAmount::<T>::get(source_asset)
+			// {
+			// 	// If the CCM's principal requires a swap and is non-zero,
+			// 	// then the principal swap amount must be above minimum swap amount required.
+			// 	return Err(CcmFailReason::PrincipalSwapAmountTooLow)
+			// }
```
Comment (dandanlen) [inline state-chain/pallets/cf-swapping/src/lib.rs:706]: Ditto, can delete.

**Review by dandanlen — COMMENTED** [review]
LGTM, just a few nitty comments.

**Pull request merged**